src/powersystem.jl

```diff
@@ -36,42 +35,44 @@ struct PowerNetwork
 
     Sbus::Array{Complex{Float64}}
 
-    function PowerNetwork(datafile::String)
-        data = Parse.parse_raw(datafile)
+    function PowerNetwork(datafile::String, data_format::Int64=0)
```

Comment:
Here, I would be in favor of using multiple dispatch on the `data_format`, and have the following signature there: 
`PowerNetwork(datafile::String, data_format::Format)`

An idea would be to merge the two submodules `ParsePSSE` and `ParseMAT` into a single module `Parser`, and implement in parser:
```julia
abstract type AbstractFormat end

struct MatpowerFormat <: AbstractFormat end
struct PSSEFormat <: AbstractFormat end

```

and then dispatch the parsing on the `AbstractFormat` type. E.g. by replacing
```julia
ParsePSSE.parse_raw(datafile)
ParseMAT.parse_mat(datafile)
```
by 
```julia
data_raw = Parser.parse(MatPowerFormat(), datafile)
data_raw = Parser.parse(PSSEFormat(), datafile)
```

That would allow greater extensibility IMO. But I think this is not blocking for this PR. Just putting this comment here to open the discussion.

Reply:
Yes, you're right. This PR is mostly gluing what we had for RAW datafiles and the Matpower format.

While I did not do a clean job I think I've modularized enough so we can refactor this without messing with the rest of the code so we can get back to this after the RGM.